chain-signatures/node/src/protocol/signature.rs

```diff
@@ -93,6 +100,65 @@ impl SignQueue {
         self.len() == 0
     }
 
+    fn contains(&self, sign_id: &SignId) -> bool {
+        self.my_requests
+            .iter()
+            .any(|req| req.indexed.id == *sign_id)
+            || self.other_requests.contains_key(sign_id)
+    }
+
+    fn organize_request(
+        &self,
+        threshold: usize,
+        stable: &[Participant],
+        indexed: IndexedSignRequest,
+        reorganize: bool,
+    ) -> SignRequest {
+        let sign_id = indexed.id.clone();
+        // NOTE: reorganize, will use the same entropy for reorganizing the participants. The only
+        // thing that would be different is the passed in stable participants.
+        let mut rng = StdRng::from_seed(indexed.args.entropy);
+        let subset = stable.iter().copied().choose_multiple(&mut rng, threshold);
```

Comment:
That's reasonable as a iterative improvement.

Long-term, shouldn't we try different subset even of the same stable set?

Reply:
yeah, I think we can move to reorganizing the same stable set later, but requires adding in an attempt counter just to change the entropy